### PR TITLE
chore: rename nodereport to node-report

### PR DESCRIPTION
**Pull request by gdams**
This changes nodereport to node-report

### CODE REVIEW

**Comment by rnchamberlain** [comment]
@gdams the hyphen is not allowed in C macro names, so the project won't build as-is. For example, NODEREPORT_VERSION will need to change to perhaps NODE_REPORT_VERSION, not node-report_VERSION. Ditto the environment variables.

File: binding.gyp [hunk 1]
```diff
@@ -21,7 +21,7 @@
         }],
       ],
       "defines": [
-        'NODEREPORT_VERSION="<!(node -p \"require(\'./package.json\').version\")"'
+        'node-report_VERSION="<!(node -p \"require(\'./package.json\').version\")"'
```
Comment (richardlau) [inline binding.gyp]: Please either revert this, or upper case. I don't believe `-` is a valid character for C/C++ macro names.

File: CHANGES.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 2016-12-12, Version 1.0.7
 =========================
 
- * Fix version reporting in NodeReport section (Richard Lau)
+ * Fix version reporting in node-report section (Richard Lau)
```
Comment (richardlau) [inline CHANGES.md]: I don't think we should rename in the history -- It was correct at the time.

File: CHANGES.md [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
  * Fix for failure in fatal error (OOM) test (Richard Chamberlain)
 
- * Add support for nodereport on AIX (Richard Chamberlain)
+ * Add support for node-report on AIX (Richard Chamberlain)
```
Comment (richardlau) [inline CHANGES.md]: I don't think we should rename in the history -- It was correct at the time.

File: LICENCE.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
 The MIT License (MIT)
 =====================
 
-Copyright (c) 2016 nodereport contributors
+Copyright (c) 2016 node-report contributors
 --------------------------------------------------
 
-*nodereport contributors listed at <https://github.com/nodejs/nodereport/blob/master/AUTHORS>*
+*node-report contributors listed at <https://github.com/nodejs/node-report/blob/master/AUTHORS>*
```
Comment (richardlau) [inline LICENCE.md]: If we're updating this file, https://github.com/nodejs/nodereport/blob/master/AUTHORS no longer exists (it was removed by #13).

File: README.md [hunk 5]
```diff
@@ -14,87 +14,87 @@ Supports Node.js v4, v6 and v7 on Linux, MacOS, Windows and AIX.
...
-export NODEREPORT_FILENAME=stdout|stderr|<filename>
-export NODEREPORT_DIRECTORY=<full path>
-export NODEREPORT_COREDUMP=yes|no
-export NODEREPORT_VERBOSE=yes|no
+export node-report_EVENTS=exception+fatalerror+signal+apicall
+export node-report_SIGNAL=SIGUSR2|SIGQUIT
+export node-report_FILENAME=stdout|stderr|<filename>
+export node-report_DIRECTORY=<full path>
+export node-report_COREDUMP=yes|no
+export node-report_VERBOSE=yes|no
```
Comment (richardlau) [inline README.md]: We want to keep the environment variables case (upper), probably without the hyphen.

File: index.js [hunk 6]
```diff
@@ -1,9 +1,9 @@
-// Main module entry point for nodereport
+// Main module entry point for node-report
 
 const api = require('./api');
 
-// NODEREPORT_EVENTS env var overrides the defaults
-const options = process.env.NODEREPORT_EVENTS || 'exception+fatalerror+signal+apicall';
+// node-report_EVENTS env var overrides the defaults
+const options = process.env.node-report_EVENTS || 'exception+fatalerror+signal+apicall';
```
Comment (richardlau) [inline index.js]: We want to keep the environment variables case (upper), probably without the hyphen.

File: src/node_report.cc [hunk 7]
```diff
@@ -93,15 +93,15 @@ static struct tm loadtime_tm_struct; // module load time
...
  *   Core dump yes/no switch
  *   Trigger signal selection
- *   NodeReport filename
- *   NodeReport directory
+ *   node-report filename
+ *   node-report directory
  *   Verbose mode
  ******************************************************************************/
-unsigned int ProcessNodeReportEvents(const char* args) {
+unsigned int ProcessProcessNode_ReportEvents(const char* args) {
```
Comment (richardlau) [inline src/node_report.cc]: Extra `Process`. Doesn't match the call in `src/module.cc`.

**Comment by richardlau** [comment]
I'd like a second opinion on the function name changes in the C++ code -- I'm finding it much harder to read. When I see `ProcessNode_ReportCoreSwitch` I'm mentally parsing the underscore as a separator so I read it as `ProcessNode` and `ReportCoreSwitch`.

File: README.md [hunk 8]
```diff
@@ -14,87 +14,87 @@ Supports Node.js v4, v6 and v7 on Linux, MacOS, Windows and AIX.
...
 by sending a USR2 signal to a Node.js process (Linux/MacOS only).
 
-A NodeReport can also be triggered via an API call from a JavaScript
+A node-report can also be triggered via an API call from a JavaScript
 application.
 
 ```js
-var nodereport = require('nodereport');
-nodereport.triggerReport();
+var node-report = require('node-report');
```
Comment (sam-github) [inline README.md]: this is syntactically invalid javascript, you need to be more discerning in your search and replace. revert it to the original.

File: README.md [hunk 9]
```diff
@@ -14,87 +14,87 @@ Supports Node.js v4, v6 and v7 on Linux, MacOS, Windows and AIX.
...
 Configuration on module Initialization is also available via environment variables:
 
 ```bash
-export NODEREPORT_EVENTS=exception+fatalerror+signal+apicall
-export NODEREPORT_SIGNAL=SIGUSR2|SIGQUIT
-export NODEREPORT_FILENAME=stdout|stderr|<filename>
-export NODEREPORT_DIRECTORY=<full path>
-export NODEREPORT_COREDUMP=yes|no
-export NODEREPORT_VERBOSE=yes|no
+export NODE_REPORT_EVENTS=exception+fatalerror+signal+apicall
```
Comment (sam-github) [inline README.md]: These don't have to be changed. Its the module name that is punctuated, we don't have to punctuate the env var names

File: README.md [hunk 10]
```diff
@@ -14,87 +14,87 @@ Supports Node.js v4, v6 and v7 on Linux, MacOS, Windows and AIX.
...
+export NODE_REPORT_DIRECTORY=<full path>
+export NODE_REPORT_COREDUMP=yes|no
+export NODE_REPORT_VERBOSE=yes|no
 ```
 
 ## Examples
 
-To see examples of NodeReports generated from these events you can run the
-demonstration applications provided in the nodereport github repository. These are
+To see examples of node-reports generated from these events you can run the
```
Comment (sam-github) [inline README.md]: "of reports generated"

File: README.md [hunk 11]
```diff
@@ -14,87 +14,87 @@ Supports Node.js v4, v6 and v7 on Linux, MacOS, Windows and AIX.
...
+demonstration applications provided in the node-report github repository. These are
```
Comment (sam-github) [inline README.md]: "in the github repository"

File: README.md [hunk 12]
```diff
@@ -14,87 +14,87 @@ Supports Node.js v4, v6 and v7 on Linux, MacOS, Windows and AIX.
...
 Node.js applications which will prompt you to trigger the required event.
 
-1. `api.js` - NodeReport triggered by JavaScript API call.
-2. `exception.js` - NodeReport triggered by unhandled exception.
-3. `fatalerror.js` - NodeReport triggered by fatal error on JavaScript heap out of memory.
-4. `loop.js` - looping application, NodeReport triggered using kill `-USR2 <pid>`.
+1. `api.js` - node-report triggered by JavaScript API call.
```
Comment (sam-github) [inline README.md]: for all `s/node-report/report/`

File: binding.gyp [hunk 13]
```diff
@@ -21,7 +21,7 @@
         }],
       ],
       "defines": [
-        'NODEREPORT_VERSION="<!(node -p \"require(\'./package.json\').version\")"'
+        'NODE_REPORT_VERSION="<!(node -p \"require(\'./package.json\').version\")"'
```
Comment (sam-github) [inline binding.gyp]: can revert, IMO

File: demo/api_call.js [hunk 14]
```diff
@@ -1,5 +1,5 @@
-// Example - generation of NodeReport via API call
-var nodereport = require('nodereport');
+// Example - generation of node-report via API call
```
Comment (sam-github) [inline demo/api_call.js]: generation of report via API call

File: demo/api_call.js [hunk 15]
```diff
@@ -1,5 +1,5 @@
...
+var node-report = require('node-report');
```
Comment (sam-github) [inline demo/api_call.js]: revert, this is syntactically invalid

File: demo/api_call.js [hunk 16]
```diff
@@ -8,13 +8,13 @@ function my_listener(request, response) {
   switch(count++) {
   case 0:
     response.writeHead(200,{"Content-Type": "text/plain"});
-    response.write("\nRunning NodeReport API demo... refresh page to trigger NodeReport");
+    response.write("\nRunning node-report API demo... refresh page to trigger node-report");
```
Comment (sam-github) [inline demo/api_call.js]: "Running API demo... refresh page to trigger report"

File: demo/api_call.js [hunk 17]
```diff
@@ -8,13 +8,13 @@ function my_listener(request, response) {
...
     response.end();
     break;
   case 1:
     response.writeHead(200,{"Content-Type": "text/plain"});
-    // Call the nodereport module to trigger a NodeReport
-    var filename = nodereport.triggerReport();
+    // Call the node-report module to trigger a node-report
```
Comment (sam-github) [inline demo/api_call.js]: "// Excplicit call to trigger a report"

File: demo/api_call.js [hunk 18]
```diff
@@ -8,13 +8,13 @@ function my_listener(request, response) {
...
+    var filename = node-report.triggerReport();
```
Comment (sam-github) [inline demo/api_call.js]: invalid, revert

File: demo/exception.js [hunk 19]
```diff
@@ -1,5 +1,5 @@
-// Example - generation of NodeReport on uncaught exception
-require('nodereport').setEvents("exception");
+// Example - generation of node-report on uncaught exception
```
Comment (sam-github) [inline demo/exception.js]: "generation of report on"

File: demo/exception.js [hunk 20]
```diff
@@ -8,7 +8,7 @@ function my_listener(request, response) {
   switch(count++) {
   case 0:
     response.writeHead(200,{"Content-Type": "text/plain"});
-    response.write("\nRunning NodeReport exception demo... refresh page to cause exception (application will terminate)");
+    response.write("\nRunning node-report exception demo... refresh page to cause exception (application will terminate)");
```
Comment (sam-github) [inline demo/exception.js]: "Running exception demo"

File: demo/fatalerror.js [hunk 21]
```diff
@@ -1,5 +1,5 @@
-// Example - generation of Nodereport on fatal error (Javascript heap OOM)
-require('nodereport').setEvents("fatalerror");
+// Example - generation of node-report on fatal error (Javascript heap OOM)
```
Comment (sam-github) [inline demo/fatalerror.js]: "generation of report on"

Comment (sam-github) [inline demo/fatalerror.js]: I won't keep mentioning this. A report is a "report" not a "NodeReport" or a "node-report".

File: index.js [hunk 22]
```diff
@@ -1,9 +1,9 @@
-// Main module entry point for nodereport
+// Main module entry point for node-report
```
Comment (sam-github) [inline index.js]: "// main module entry point"

File: index.js [hunk 23]
```diff
@@ -1,9 +1,9 @@
...
 
 const api = require('./api');
 
-// NODEREPORT_EVENTS env var overrides the defaults
-const options = process.env.NODEREPORT_EVENTS || 'exception+fatalerror+signal+apicall';
+// NODE_REPORT_EVENTS env var overrides the defaults
```
Comment (sam-github) [inline index.js]: revert, unnecessary to change this

File: src/module.cc [hunk 24]
```diff
@@ -1,7 +1,7 @@
 #include "node_report.h"
 #include <nan.h>
 
-namespace nodereport {
+namespace node-report {
```
Comment (sam-github) [inline src/module.cc]: invalid, revert to original

Comment (sam-github) [inline src/module.cc]: I think every change in this file pretty much, except maybe some comments, can be reverted as unnecessary (and won't compile).

File: src/module.cc [hunk 25]
```diff
@@ -241,26 +241,26 @@ static int StartWatchdogThread(void* (*thread_main)(void* unused)) {
   pthread_sigmask(SIG_SETMASK, &sigmask, nullptr);
   pthread_attr_destroy(&attr);
   if (err != 0) {
-    fprintf(stderr, "nodereport: StartWatchdogThread pthread_create() failed: %s\n", strerror(err));
+    fprintf(stderr, "node-report: StartWatchdogThread pthread_create() failed: %s\n", strerror(err));
```
Comment (sam-github) [inline src/module.cc]: this change is one of the few that should stay, because it names the module

File: src/module.cc [hunk 26]
```diff
@@ -241,26 +241,26 @@ static int StartWatchdogThread(void* (*thread_main)(void* unused)) {
...
 }
 
-// Watchdog thread implementation for signal-triggered NodeReport
+// Watchdog thread implementation for signal-triggered node-report
 inline void* ReportSignalThreadMain(void* unused) {
   for (;;) {
     uv_sem_wait(&report_semaphore);
-    if (nodereport_verbose) {
-      fprintf(stdout, "nodereport: signal %s received\n", node::signo_string(report_signal));
+    if (node-report_verbose) {
```
Comment (sam-github) [inline src/module.cc]: don't need this line

File: src/module.cc [hunk 27]
```diff
@@ -241,26 +241,26 @@ static int StartWatchdogThread(void* (*thread_main)(void* unused)) {
...
+      fprintf(stdout, "node-report: signal %s received\n", node::signo_string(report_signal));
```
Comment (sam-github) [inline src/module.cc]: do need this line. etc.

**Comment by richardlau** [comment]
Superseded by #43.

**Pull request closed without merging**